### ISSUE TITLE
[BUG] Unable to run set-pnpsite -Disablesharingfornonowners  

### ISSUE THREAD

**Issue by reshmee011** [pnp/powershell#3711]
### Notice
Many bugs reported are actually related to the PnP Framework which is used behind the scenes. Consider carefully where to report an issue:

1. **Are you using ```Invoke-PnPSiteTemplate``` or ```Get-PnPSiteTemplate```**? The issue is most likely related to the Provisioning Engine. The Provisioning engine is _not_ located in the PowerShell repo. Please report the issue here: https://github.com/pnp/pnpframework/issues.
1. **Is the issue related to the cmdlet itself, its parameters, the syntax, or do you suspect it is the code of the cmdlet that is causing the issue?** Then please continue reporting the issue in this repo.
1. **If you think that the functionality might be related to the underlying libraries that the cmdlet is calling** (We realize that might be difficult to determine), please first double check the code of the cmdlet, which can be found here: https://github.com/pnp/powershell/tree/master/src/Commands. If related to the cmdlet, continue reporting the issue here, otherwise report the issue at https://github.com/pnp/pnpframework/issues

### Reporting an Issue or Missing Feature
Please confirm what it is that your reporting

### Expected behavior 
The site setting disablesharingfornonowners is correctly processed

### Actual behavior 

<img width="709" alt="image" src="https://github.com/pnp/powershell/assets/7693852/a43d855d-5dfa-4e67-a948-faed19f4aa6a">

Error message is displayed
Set-PnPSite: The type of data at position 3330 is different than the one expected.

### Steps to reproduce behavior
connect-pnpOnline -Url https://reshmeeauckloo.sharepoint.com/sites/testclone2 -interactive
set-pnpsite -Disablesharingfornonowners

### What is the version of the Cmdlet module you are running?
2.3.37-nightly

### Which operating system/environment are you running PnP PowerShell on?
- [ ] Windows

I can reproduce the same error using SPO PowerShell
```PowerShell
$AdminSiteURL= "https://reshmeeauckloo-admin.sharepoint.com"
$SiteURL= "https://reshmeeauckloo.sharepoint.com/sites/testclone2"
  
#Connect to SharePoint Online services
Connect-SPOService -url $AdminSiteURL
 
#Disable Sharing for Non-Owners
Set-SPOSite -Identity $SiteURL -DisableSharingForNonOwners
```
<img width="630" alt="image" src="https://github.com/pnp/powershell/assets/7693852/37ade4b9-407b-4370-97e5-12248ab73450">

Is it something to raise with Microsoft team?

**Comment by veronicageek** [comment]
Hi @reshmee011 - I guess we just consume from CSOM so if SP Management Shell has the same issue, we're not going to be able to do anything

**Comment by reshmee011** [comment]
@veronicageek : I thought so too, is it worth raising it with Microsoft ?

**Comment by veronicageek** [comment]
You could (maybe Vesa) and once fixed, it'll be fixed here.

**Comment by LastGunslinger** [comment]
Can this issue please be re-opened? I have been talking with Microsoft support since this issue started, and they have made it pretty clear that whatever they changed in this case was intentional, which means that they likely don't plan on going back to fix this implementation which they broke. On their end, they released version 16.0.24510.12000 of their Microsoft.Online.SharePoint.PowerShell module specifically to address this problem. The fact that they have updated their module with a new version instead of making a fix on the back end like they have before in these instances is very telling. This is what I was told by their support engineer after going back and forth with them for several weeks:

> I believe there is a new version of the Microsoft.Online.SharePoint.PowerShell module
> We now have version 16.0.24510.12000, and this version resolved this issue

I expressed to their support that we use the PnP.PowerShell module because it allows us to use PS7, while theirs is locked to PS5.1, but their only response was that PS7 isn't officially supported for any of their SharePoint modules.

If Microsoft doesn't plan to fix this, then it's very likely that this will also need to addressed within this module. I asked the support tech what had changed on the back end in the hopes that I could open a PR to fix it here, but of course they wouldn't tell me. In either case, until this is fixed, the following cmdlets are broken for me:

Get-PnPExternalUser
Remove-PnPExternalUser
Set-PnPTenantSite (specifically when using the DisableSharingForNonOwners flag)

**Comment by reshmee011** [comment]
@LastGunslinger , I have checked the cmdlets with PnP version 2.3.0 and 2.3.37 and could not reproduce the errors anymore. May be MS has not deployed the fix onto your tenant.
